ObliviousHarmony/vscode-php-codesniffer
# Added `Automatic` Standard Option

### All Submissions:


### Changes proposed in this Pull Request:

This pull request adds a new `Automatic` option for `phpCodeSniffer.standard`. When this new option is used the extension will attempt to find one of the configuration file types that we've defined. It will traverse from the document's directory to the workspace root seeking out an appropriate file.

As part of this work I also went ahead and refactored the way that the standard works a little bit. There's no longer a need to check for type strings and I've removed the `StandardType` enum entirely.

Closes #59.

### How to test the changes in this Pull Request:

1. Set `phpCodeSniffer.standard` to `Automatic` in the settings.
2. Create a workspace that has both a root `phpcs.xml` file and a `phpcs.xml` file in a nested directory.
3. Create a PHP file in the nested directory and make sure it violates the nested coding standard but not the one at the root.
4. Delete the nested coding standard and the errors should go away.
5. Rename the root file to `.phpcs.xml` and verify that it still works.
